### PR TITLE
Generate docs content to Pages

### PR DESCRIPTION
**Pull request by fabriguespe**
Fixed SDK docs [issues](https://github.com/xmtp/xmtp-ios/issues) while doing docs deployment

- [Preview](https://xmtp.github.io/xmtp-ios/)

```yml
name: Deploy docs content to GitHub Pages

on:
  push:
    branches: ["main", "docs", "docs_build_v1"]
  workflow_dispatch:

permissions:
  contents: read
  pages: write
  id-token: write

concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: macos-12
    steps:
      - uses: actions/checkout@v3

      - name: Set up Ruby
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: 3.0

      - name: Cache Ruby gems
        uses: actions/cache@v2
        with:
          path: vendor/bundle
          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
          restore-keys: |
            ${{ runner.os }}-gems-

      - name: Install Jazzy
        run: |
          bundle config path vendor/bundle
          bundle add jazzy

      - name: Generate documentation
        run: bundle exec jazzy --output ./docs --theme=fullwidth

      - name: Setup Pages
        uses: actions/configure-pages@v3

      - name: Upload artifact
        uses: actions/upload-pages-artifact@v1
        with:
          path: "./docs"

      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v2
```

To run this locally:

```
gem install jazzy
jazzy --output ./docs --theme=fullwidth
```

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: Deploy docs content to Pages 2
+
+on:
+  push:
+    branches: ["docs_build_v1"]
```
Comment (nplasterer) [inline .github/workflows/docs.yml:5]: We probably want this to actually normally happen on pushes to main 

```suggestion
    branches: ["main", "docs_build_v1"]
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: Deploy docs content to Pages 2
```
Comment (nplasterer) [inline .github/workflows/docs.yml:1]:
```suggestion
name: Deploy docs content to GitHub Pages
```

File: Sources/XMTP/Client.swift [hunk 3]
```diff
@@ -42,7 +42,7 @@ public struct ClientOptions {
 	}
 }
 
-/// Client is the entrypoint into the XMTP SDK.
+/// Client is the entrypoint into the XMTP SDK.  v2
```
Comment (nplasterer) [inline Sources/XMTP/Client.swift:45]:
```suggestion
/// Client is the entrypoint into the XMTP SDK.
```

File: Gemfile [hunk 4]
```diff
@@ -0,0 +1,2 @@
+source 'https://rubygems.org'
+gem 'jazzy'
```
Comment (nplasterer) [inline Gemfile:2]: I don't know enough about iOS development to know if this would be an issue cc @nakajima

**Review by nplasterer — COMMENTED** [review]
Nice! Seems to be working. I'd suggest deleting the old docs deployment action so we don't have two running https://github.com/xmtp/xmtp-ios/blob/main/.github/workflows/build-docs.yml

File: Gemfile (hunk 4 above)
Comment (nakajima) [reply Gemfile:2]: Seems fine to me.

**Review by nakajima — APPROVED** [review]

**Pull request merged**